dpypelines/pipeline/generic_file_ingress_v1.py

```diff
@@ -0,0 +1,199 @@
+            data={"required_file_patterns": required_file_patterns},
+        )
+    except Exception as err:
+        logger.error(
+            "Error occurred when getting required file patterns",
+            err,
+            data={
+                "pipeline_config": pipeline_config,
+            },
+        )
+        de_notifier.failure()
+        raise err
+
+    # Check for the existence of each required file
+    for required_file in required_file_patterns:
+        try:
+            if not local_store.has_lone_file_matching(required_file):
+                email_content = file_not_found_email(required_file)
+                email_client.send(
+                    submitter_email, email_content.subject, email_content.message
+                )
+                logger.info(
+                    "Email sent to submitter about missing required file",
+                    data={
+                        "submitter_email": submitter_email,
+                        "required_file": required_file,
+                    },
+                )
+                de_notifier.failure()
+                raise FileNotFoundError(
+                    f"Could not find file matching pattern {required_file}"
+                )
+        except Exception as err:
+            files_in_directory = local_store.get_file_names()
+            logger.error(
+                "Error occurred when looking for required file",
+                err,
+                data={
+                    "required_file": required_file,
+                    "required_file_patterns": required_file_patterns,
+                    "files_in_directory": files_in_directory,
+                    "pipeline_config": pipeline_config,
+                },
+            )
+            de_notifier.failure()
+            raise err
+
+    # Allow DE's to skip the upload to s3 part of the pipeline while
+    # developing code locally.
+    skip_data_upload = os.environ.get("SKIP_DATA_UPLOAD", False)
+    if skip_data_upload is not False:
+        try:
+            skip_data_upload = str_to_bool(skip_data_upload)
+        except Exception as err:
+            logger.error(
+                "Unable to cast SKIP_DATA_UPLOAD to boolean",
+                err,
+                data={"value": skip_data_upload},
+            )
+            de_notifier.failure()
+            raise err
+
+    logger.info(
+        "skip_data_upload set from SKIP_DATA_UPLOAD env var",
+        data={"value": skip_data_upload},
+    )
+
+    if skip_data_upload is not True:
+
+        # Upload output files to Upload Service
+        try:
+            # Create UploadClient from upload_url
+            upload_client = UploadServiceClient(upload_url)
+            for required_file in required_file_patterns:
+                if "manifest.json" in required_file:
+                    continue
+                elif ".xml" in required_file:
+                    upload_client.upload_new_sdmx(files_dir)
```

Comment:
files_dir is the entire directory, you need to be telling it to upload the **specific** file that matches the required_file that's not manifest.json.

you'll need to save the file matching the pattern to get the path, then pass that path into the upload method, see here: https://github.com/ONSdigital/dp-data-pipelines/blob/6f13dd3c7e992cf4913d7d8a31fc41cc124cdce0/dpypelines/pipeline/dataset_ingress_v1.py#L424

Reply:
done.